src/leveldb/db/log_reader.cc

```diff
@@ -39,7 +39,6 @@ bool Reader::SkipToInitialBlock() {
 
   // Don't search a block if we'd be in the trailer
   if (offset_in_block > kBlockSize - 6) {
-    offset_in_block = 0;
```

Comment:
leveldb is maintained as subtree, we usually don't change leveldb locally. Better way to do it is to submit a patch upstream or check if upstream has a fix already.

Reply:
Fair point, I'll have a look and take the level DB changes out of this commit.